src/ATen/function_wrapper.py

```diff
@@ -75,6 +75,22 @@
     return static_cast<Type*>(this)->${method_prefix}${api_name}(${sparse_actuals});
 }""")
 
+WRAP_DIM_GEN = CodeTemplate("""\
```

Comment:
I think the logic here is good, but can we put the body of this function in a function? When you are generating a lot of code, keeping the size of the generated code small is important for maintainability.

Reply:
Sure, will do.  One question that came up based on other review comments -- should we support dim specification on scalars?
Only 0 or -1 would be allowed, which is what numpy allows.  What do you think?

Reply:
What do you mean by dim specifications on scalars? How does that manifest?

Reply:
Oh I was confused by the terminology because there is also the at::Scalar class. I think a zero-dim tensor should throw an error on `zero_dim.prod(any_number)` because the invariant `0 <= any_number < zero_dim.dim()` fails.

Reply:
is there a term you had in mind for tensors of this kind (for which .dim() returns 0)?